core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -590,12 +593,59 @@ private void checkMerkleRoot() throws VerificationException {
         }
     }
 
+    private void checkSegwitCommit() throws VerificationException {
+        if (transactions == null)
+            throw new VerificationException("No transactions in block");
+        final Transaction coinBase = transactions.get(0);
+        byte[] rootHash = null;
+        for (TransactionOutput out : coinBase.getOutputs()) {
+            final byte[] pkScript = out.getScriptBytes();
+            if (pkScript.length < 38) continue;
+            final byte[] expected = HEX.decode("6a24aa21a9ed");
```

Comment:
I'm a Haskell programmer and I'm used to this kind of statement getting memoized by the runtime or removed by the compiler. My bad. I corrected and defined a constant as suggested.